### PR TITLE
[FLINK-34679][cdc] add doc under core-concept.

### PR DESCRIPTION
**Pull request by LYanquan**
A follow up of https://github.com/apache/flink-cdc/pull/3146 to complete the docs.

### CODE REVIEW

**Comment by LYanquan** [comment]
@leonardBang  @PatrickRen PTAL.

File: docs/content/docs/core-concept/route.md [hunk 1]
```diff
@@ -23,3 +23,52 @@ KIND, either express or implied.  See the License for the
...
+if synchronize the table `web_order` in the database `mydb` to a Doris table `ods_web_order`, we can use this yaml file to define this route：
+
+```yaml
+route:
+    source-table: mydb.web_order
+    sink-table: mydb.ods_web_order
+    description: sync table to one destination table with given prefix ods_
+```
+
+## many to one
```
Comment (leonardBang) [inline docs/content/docs/core-concept/route.md:52]: Route multiple Data Source tables to one Data Sink table

File: docs/content/docs/core-concept/route.md [hunk 2]
```diff
@@ -23,3 +23,52 @@ KIND, either express or implied.  See the License for the
...
+What's more, if you want to synchronize the sharding tables in the database `mydb` to a Doris table `ods_web_order`, we can use this yaml file to define this route：
+```yaml
+route:
+    source-table: mydb\.*
+    sink-table: mydb.ods_web_order
+    description: sync sharding tables to one destination table
+```
+
+## many rules
```
Comment (leonardBang) [inline docs/content/docs/core-concept/route.md:61]: Complex Route via combining route rules

**Review by leonardBang — COMMENTED** [review]
Thanks @LYanquan for the contribution, I left some comments

**Comment by LYanquan** [comment]
address it.

File: docs/content/docs/core-concept/data-sink.md [hunk 3]
```diff
@@ -23,3 +23,28 @@ KIND, either express or implied.  See the License for the
...
+A Data Sink can write to multiple tables simultaneously.
+
+# Parameters
+To describe a data sink, the follows are required:
+
+| parameter                   | meaning                                                                               | optional/required |
+|-----------------------------|---------------------------------------------------------------------------------------|-------------------|
+| type                        | The type of the sink, such as doris or starrocks.                                     | required          |
+| name                        | The name of the sink, which is user-defined (a default value provided).               | optional          |
+| other custom configurations | custom configurations for the sink to specify the connection config and table config. | optional          |
```
Comment (leonardBang) [inline docs/content/docs/core-concept/data-sink.md:38]:
```suggestion
| configurations of Data Sink |  Configurations to build the Data Sink e.g. connection configurations and sink table properties. | optional          |
```

Comment (LYanquan) [reply docs/content/docs/core-concept/data-sink.md]: address it.

File: docs/content/docs/core-concept/transform.md [hunk 4]
```diff
@@ -23,3 +23,10 @@ KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->
+
+# Definition
+**Transform** module helps users delete and expand data columns based on the data columns in the table.      
+What's more, it also helps users filter some unnecessary data during the synchronization process.
+
+# Example
+This feature has not been implemented now.
```
Comment (leonardBang) [inline docs/content/docs/core-concept/transform.md:32]:
```suggestion
This feature will support soon.
```

**Review by leonardBang — APPROVED** [review]
Thanks @LYanquan for the update, LGTM

**Pull request merged**